### PR TITLE
Modernize dependencies and structure

### PR DESCRIPTION
**Pull request by Redth**

### CODE REVIEW

**Comment by casz** [comment]
I'll have a look at it

**Comment by Redth** [comment]
@casz not sure why it's failing on the Cake.Recipes stuff... Perhaps we can discuss in gitter?

**Comment by gep13** [comment]
@Redth am I right in saying that you have got the Cake.Recipe related build issues fixed now?

**Comment by Redth** [comment]
@gep13 yes! And after this last commit I think this is good to go.

**Comment by gep13** [comment]
@Redth sweet!  Thanks for helping with this addin, appreciate it!  Feel free to reach out if there is anything you need help with!

**Comment by casz** [comment]
May I ask why you did d72d26311ea6ca70d8065518100b4f071df3755f moving the tests out of the solution?

**Comment by Redth** [comment]
@casz yes, because cake.recipe uses dotnet msbuild which didn't play nicely with the xamarin app head projects (the xunit and nunit runner app projects) when building.

File: src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj [hunk 1]
```diff
@@ -1,136 +1,43 @@
...
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'UnitTesting|AnyCPU' ">
-    <Optimize>false</Optimize>
-    <OutputPath>bin\UnitTesting</OutputPath>
-    <WarningLevel>4</WarningLevel>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (casz) [inline src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj]: Any reason to only target netcoreapp2.0?

Comment (Redth) [reply src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj]: Any particular reason to target anything else?  Are you thinking netcoreapp3.0 ?

Comment (Redth) [reply src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj]: I don't have a particular reason other than it's what I've used in my other cake addins and it got copy pasta'd over ;)

Comment (casz) [reply src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj]: Why not just multi target both netcore2.1 and netcore3.1 as they are both in LTS status?

Comment (Redth) [reply src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj]: Sure, pushed changes to multitarget 2.1 and 3.1 just now.

Comment (Redth) [reply src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj]: Ahh so, AppVeyor doesn't want to handle 3.1 at this point.  Honestly I don't see a ton of value adding 3.1 support for the tests right now given it's a netstandard project it's testing.  I made it netcoreapp2.1 instead and I think that's fine.

Comment (casz) [reply src/Cake.AppleSimulator.Tests/Cake.AppleSimulator.Tests.csproj]: On AppVeyor it sohuld use the Visual Studio 2019 image than 3.1 support should work just fine.

**Comment by Redth** [comment]
@casz anything else?

**Comment by casz** [comment]
LGTM however I am somewhat concerned about the `54 Warning(s)` 😱

**Pull request merged**